PirateInterpreter/Values/Variable.cs

```diff
@@ -9,35 +9,12 @@ public class Variable : BaseValue, IValue
     public Variable(string value, ILogger logger, InterpreterFactory interpreterFactory): base(value, logger)
     {
         InterpreterFactory = interpreterFactory;
-        Value = GetValue(value);
-    }
-
-    public object GetValue(string value)
-    {
-        var resultNode = SymbolTable.Instance(Logger).Get(value);
-        if (resultNode is not IValueNode)
-        {
-
-            var interpreter = InterpreterFactory.GetInterpreter(resultNode, Logger);
-            var result = interpreter.VisitNode();
-            if (result.Count == 1)
-            {
-                return result[0].Value;
-            }
-            else
-            {
-                throw new Exception("Variable is not a single value");
-            }
-        }
-        else
-        {
-            var valueNode = (IValueNode)resultNode;
-            return valueNode.Value.Value;
-        }
+        Value = SymbolTable.Instance(Logger).Get(value).Value;
     }
 
     public override BaseValue OperatedBy(Token _operator, BaseValue other)
     {
+        Logger.Log($"Variable {Value.ToString()}, {Value.GetType()} is being operated by {other.ToString()}, {other.GetType()} with {_operator.ToString()}", this.GetType().Name, LogType.INFO);
         switch (Value.GetType())
         {
             case Type intType when Value.GetType() == typeof(int):
```

Comment:
Maybe asked before but isn't it possible to write it like this:
Case Type intType when Value is int32:

Reply:
![image](https://user-images.githubusercontent.com/98022003/202192702-f98aec46-3feb-4f64-9598-6f1381a66d98.png)